src/main/java/com/google/codeu/data/Datastore.java

```diff
@@ -114,5 +117,30 @@ public void addMessage(List<Message> messages, Entity entity, String user) {
 
   return messages;
  }
+  /** 
+   * Gets all the users
+   * 
+   * @return a set of users
+   */
+  public Set<String> getUsers(){
+    Set<String> users = new HashSet<>();
+    Query query = new Query("Message");
+    PreparedQuery results = datastore.prepare(query);
+    for(Entity entity : results.asIterable()) {
+      users.add((String) entity.getProperty("user"));
+    }
+    return users;
+  }
 
+  /** Returns the total number of messages for all users. */
+  public int getTotalMessageCount(){
+    Query query = new Query("Message");
+    PreparedQuery results = datastore.prepare(query);
+    return results.countEntities(FetchOptions.Builder.withLimit(1000));
```

Comment:
The 1000 here could be a const, like this
private static final long MAX_ENTITIES = 1000;